docs/pages/blog/base-ui-2024-plans.md

```diff
@@ -0,0 +1,105 @@
+| Radio Group            |                [View the GitHub issue&nbsp;→](https://github.com/mui/material-ui/issues/38038) |
+| Tooltip                |                [View the GitHub issue&nbsp;→](https://github.com/mui/material-ui/issues/38045) |
+
+## Improved customization API
+
+Currently, Base UI components can be customized to your heart's content using the `slots` and `slotProps` props.
+(Read more about them in the "[Overriding component structure](/base-ui/guides/overriding-component-structure/)" guide.)
+
+```tsx
+// Example of the slots prop
+<Select slots={{ listbox: 'ol' }} defaultValue="First option">
+  <Option value="First option">First option</Option>
+  <Option value="Second option">Second option</Option>
+</Select>
+
+// Example of the slotProps prop
+<Badge slotProps={{ badge: { className: 'my-badge' } }} />
+```
+
+This API, while powerful, has proven to be less than ideal in some instances.
+Most notably, it's too lengthy and complicated to write and read when used with libraries such as Tailwind CSS.
+Additionally, since the `slots` and the corresponding `slotProps` are not related in terms of TypeScript types, it's possible to introduce bugs or have the compiler complain about valid code.
+
+To address these issues, we're considering adopting a new API that would assign a discrete subcomponent to each DOM node—the pattern favored by many other headless component libraries (think: `<Slider.Track />`, `<Slider.Thumb />`, etc.).
+This pattern has the potential to radically improve the customization experience, both for styles and structure.
+We're still fleshing out the details and plan to release an RFC on GitHub in the coming days—we'll update this blog post with the link when it's available.
+We know that a significant number of projects depend on the existing API, and we want to assure you that one of our top priorities is to provide a smooth migration experience.
+
+## More thorough animation support
+
+Animation is a key element for adding delight to any application.
+We've already kicked off work on animation support by releasing the [CSS Transition](/base-ui/react-transitions/#css-transition) and [CSS Animation](/base-ui/react-transitions/#css-animation) components, as well as the `useTransitionTrigger` and `useTransitionStateManager` hooks.
+They're currently available for use with the Popup, Menu, and Select, and the plan is to extend support to more components while also adding more features.
+
+{{"demo": "../../data/base/components/transitions/AllTransitionsDemo.js", "hideToolbar": true}}
+
+<p class="blog-description" style="margin: -16px 0 0 0;">The CSS Animation transition is exaggerated here for the sake of demonstration.</p>
+
+## Independent component versioning
+
+Instead of versioning the package as a whole, we're transitioning to offering each Base UI component as its own individual NPM package with its own version.
```

Comment:
Not sure if we really were considering having and offering the two options. @michaldudak & @colmtuite to confirm there.

Reply:
> Greenfield projets

It's not as simple as this imo. Independent packages are not really busy work in a greenfield project imo; you can just install the packages and it will almost always be fine. But then as your project is live in production and you need features or have bugs, you often want to just update/fix a single component and deploy the change/fix asap. You don't want to have to update 6 components at the same time, when you might not even want the changes/fixes for those other components.

> Not sure if we really were considering having and offering the two options

We could provide both options, which would be nice. But we haven't made a decision on it yet. imo, we haven't made a final decision on whether to support independent versioning at all yet. There are still things to consider.

Reply:
Okay, thank you! For pros & cons & alternatives, we can continue on the shaping page. For now, I've rephrased the paragraph a bit to say that we are **_considering_** instead of **_transitioning_**. This might be enough to leave it slightly open-ended. More details and options for the community to weigh in (if we want) can come later.

Reply:
We are considering both options: https://www.notion.so/mui-org/base-ui-Release-components-independently-a4d3c16750dd4e42b72fdfec1cc20d97

The single package has the drawback of being trickier to version. If we solve this, we may be able to publish it as well.

Reply:
> We are considering both options: https://www.notion.so/mui-org/base-ui-Release-components-independently-a4d3c16750dd4e42b72fdfec1cc20d97

Ah nice, we can move the discussion there.